### PR TITLE
[Closes #11] Add Advanced IP API documentation

### LINKED ISSUE #11: Add Advanced IP API documentation
We need to document the routes used for advanced IPs.  

Memo - Jake wrote the API calls, but you have probably used them the most.  Is this something that you can take a shot at?

### PR DESCRIPTION
**Pull request by gtenaschuk**

### CODE REVIEW

**Comment by betawaffle** [comment]
LGTM, I think.

**Comment by lucasmpb** [comment]
Looks good to me too. @gtenaschuk can you merge this and push it to the website?

File: docs/001-devices.apib [hunk 1]
```diff
@@ -243,3 +243,67 @@ Provides a listing of available services plans available to provision your devic
                 ...
               ]
             }
+
+## Device IP Address [/devices/{id}/ips]
+Assign an IP to a device, it must be reserved to my project first.
```
Comment (theunraveler) [inline docs/001-devices.apib]: I think the wording here is a little funny. Perhaps something like this instead:

> Assign an IP address to a device. The IP address must be in one of the IP ranges assigned to the device's project.

File: docs/001-devices.apib [hunk 2]
```diff
@@ -243,3 +243,67 @@ Provides a listing of available services plans available to provision your devic
...
+              "network": "75.72.66.16", // same as IP address for IPv4 || network address for IPv6
+              "address_family": 4,
+              "public": true, // true || false
+              "cidr": 32,
+              "assigned_to": { "href":"/devices/adf8cdfb-9574-4e2f-ae6f-5eaeb4aec5f7" },
+              "href": "/ips/6d1f1ffa-7912-4b78-b50d-88cc7c8ab40f"
+            }
+
+## IP Address [/ips/{id}]
+In order to see information about an IP address.
```
Comment (theunraveler) [inline docs/001-devices.apib]: Also oddly worded. Maybe something like this?

> See information about an IP address assignment.

File: docs/001-devices.apib [hunk 3]
```diff
@@ -243,3 +243,67 @@ Provides a listing of available services plans available to provision your devic
...
+
+### View an IP Address [GET]
+Returns a single IP address object.
+
++ Response 200
+
+  [IP Address][]
+
+### Unassign an IP Address [DELETE]
+Unassign an IP address record. This will remove the relationship between an IP and the device and will left the IP address available to be assigned to another device.
```
Comment (theunraveler) [inline docs/001-devices.apib]: > ... IP and the device and will left the IP address available ...

should be 

> ... IP and the device and will make the IP address available ...

"left" -> "make"

File: docs/002-projects.apib [hunk 4]
```diff
@@ -337,3 +337,94 @@ Accept or decline a transfer request.
...
+            {
+              "type": "public_ipv4", // public_ipv4 || global_ipv4
+              "quantity": 8, // How many IPv4 you want to request. Options: 1, 2, 4, 8, 16, 32, 64, 128, 256
+              "comments": "This is my comment to the Packet team" // Optional comment
+            }
+
++ Response 204
+
+## IP Reservation [/ips/{id}]
+In order to see information about an IP reservation.
```
Comment (theunraveler) [inline docs/002-projects.apib]: See the wording comments on `/devices/:id/ips` above--just remove the "In order to" part.

File: docs/002-projects.apib [hunk 5]
```diff
@@ -337,3 +337,94 @@ Accept or decline a transfer request.
...
+
+### View an IP Reservation [GET]
+Returns a single IP reservation object.
+
++ Response 200
+
+  [IP Reservation][]
+
+### Remove an IP Reservation [DELETE]
+Remove an IP reservation record. This will remove the IP reservation from the project.
```
Comment (theunraveler) [inline docs/002-projects.apib]: This is a bit redundant. Perhaps something like this?

> Remove an IP reservation from the project.

**Comment by theunraveler** [comment]
Thanks @gtenaschuk ! Just left a few comments about the wording and clarity, but looks good overall.

**Comment by gtenaschuk** [comment]
Thanks @theunraveler, working on them now

**Comment by gtenaschuk** [comment]
@theunraveler could you review it again and see if it is ready to be merged?
Thanks

**Comment by theunraveler** [comment]
Looks great--thanks again.

**Pull request merged**